syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.raft";
option java_outer_classname = "RaftProto";
option objc_class_prefix = "HLW";

package raft;

service Raft {
  rpc AppendEntries (AppendEntriesRequest) returns (AppendEntriesReply) {}
}

message ReplicatedLogEntry {
  string command = 1;
  string key = 2;
  string value = 3;
  uint64 term = 4;
}

message AppendEntriesRequest {
  uint64 prevlogindex = 1;
  repeated ReplicatedLogEntry entries = 2;
  uint64 commitindex = 3;
  uint64 term = 4;
}

message AppendEntriesReply {
  uint64 term = 1;
  bool success = 2;
}